scripts/merge-analyses.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+
+# This script merges the analysis data in the files provided as
+# arguments, and prints the merged analysis data to stdout.
+# The "target" data lines from the input files are copied to
+# the output as-is. The "source" data lines are merged such
+# that the syntax and symbol properties are unioned across
+# all input lines that have a matching (loc, pretty) tuple.
+# This ensures that for a given identifier, only a single context
+# menu item will be displayed for a given "pretty" representation,
+# and that context menu will link to all the symbols from all
+# the input files that match that.
+
+# Note that this code intentionally preserves the order of stuff
+# within each line, as much as possible. However, the lines
+# themselves get reordered, but the output can be piped through
+# `sort` as needed.
+
+from collections import OrderedDict
+import json
+import sys
+
+sourcemap = {}
```

Comment:
This should also be an OrderedDict if you want the output to be deterministic.

Reply:
I guess the sort and uniq you run on the output means this doesn't matter.

Reply:
Yeah I figured that I need to run sort and uniq on it anyway because the source lines from one platform will end up getting added to the sourcemap after all the source lines for the other platform (assuming the lines were distinct and therefore didn't get merged).